main.go

```diff
@@ -214,29 +283,42 @@ func (s *Server) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 		if len(tags) > 0 {
 			tags = " (" + tags + ")"
 		}
-		_, _ = res.Write([]byte(fmt.Sprintf(`
-<li>
-	<a href="%s">
-		%s port %d%s
-	</a>
-</li>
-		`, u, fullHostname, option.Port, tags)))
+
+		data.Results = append(data.Results, struct {
+			Url          *url.URL
+			FullHostname string
+			Port         uint16
+			Tags         string
+		}{
+			Url:          u,
+			FullHostname: fullHostname,
+			Port:         option.Port,
+			Tags:         tags,
+		})
+	}
+
+	err = tResultsList.Execute(res, data)
+	if err != nil {
+		http.Error(res, err.Error(), http.StatusInternalServerError)
+		return
 	}
 
-	_, _ = res.Write([]byte("</ul><br />"))
 	s.printQuickLinks(res, hostname)
 }
 
 func (s *Server) printHostnameTips(res http.ResponseWriter) {
-	_, _ = res.Write([]byte(`
-<p>The hostname should be in one of these formats:</p>
-<ul>
-  <li><b>ServiceName</b>.service.consul</li>
-  <li><b>PortName</b>.<b>ServiceName</b>.service.consul</li>
-  <li><b>ServiceName</b>.service.<b>DatacenterName</b>.consul</li>
-  <li><b>PortName</b>.<b>ServiceName</b>.service.<b>DatacenterName</b>.consul</li>
-</ul>
-	`))
+	_, err := fmt.Fprint(res, `
```

Comment:
I'm not sure this fixes the root issue, which is to have any HTML content that gets served be generated via `html/template` to take advantage of its correctness/safety checks.

This may be static content now, but using it here will future-proof against syntax errors & future vulnerabilities if this does become dynamic.